js/packages/teams-ai/src/Sso/OnBehalfOfCredentialAuthConfig.ts

```diff
@@ -0,0 +1,35 @@
+export type OnBehalfOfCredentialAuthConfig = {
+  /**
+   * Hostname of AAD authority.
+   */
+  authorityHost: string;
+
+  /**
+   * The client (application) ID of an App Registration in the tenant
+   */
+  clientId: string;
+
+  /**
+   * AAD tenant id
+   *
+   * @readonly
+   */
+  tenantId: string;
+} & (
+  | {
```

Comment:
question for learning: How does the `& (|{` function together in types? I don't think I've seen it before

Reply:
This should be some kind of union type. The purpose is only allowing user to define one of client secret or client certificate, but not both. Because define both parameters cause ambiguity.